app/routes/surveys/components/AdminSideBar.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import * as React from 'react';
+import styles from './surveys.css';
+import { Link } from 'react-router';
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+import type { ActionGrant } from 'app/models';
+import { ContentSidebar } from 'app/components/Content';
+
+type Props = {
+  surveyId: number,
+  deleteFunction?: number => Promise<*>,
+  push?: string => void,
+  actionGrant: ActionGrant
+};
+
+const AdminSideBar = ({
+  surveyId,
+  deleteFunction,
+  actionGrant,
+  push
+}: Props) => {
+  // const canEdit = actionGrant.includes('edit');
+  const canEdit = true;
+
+  const onConfirm = () => {
+    if (deleteFunction && push) {
+      return (
+        deleteFunction &&
```

Comment:
Isn't `push` and `deleteFunction` always supposed to be truthy (to be functions), since they are always mapped into props in the `mapDispatchToProps`?
```jsx
type Props = {
  surveyId: number,
  deleteFunction: number => Promise<*>,
  push: string => void,
  actionGrant: ActionGrant
};
```

Reply:
Looks like I messed up those Props. I intended to have `deleteFunction` as an optional prop so that not every component that wanted to use this sidebar wouldn't need to include the deleteFunction in its route and pass it down and all that.

Reply:
Yeah, that looks like the reason. Because then flow will force you to write it like this